src/frontend/src/components/registrationLink.ts

```diff
@@ -0,0 +1,36 @@
+import { html } from "lit-html";
+import { LoginFlowResult } from "../flows/login/flowResult";
+import { register } from "../flows/register";
+import { registerDisabled } from "../flows/registerDisabled";
+
+export const registrationSection = html` <div
+  class="textLink"
+  id="registerSection"
+>
+  New?
+  <button id="registerButton" class="linkStyle">
+    Create an Internet Identity Anchor.
+  </button>
+</div>`;
+
+/** Check that the current origin is not the explicit canister id or a raw url.
+ *  Explanation why we need to do this:
+ *  https://forum.dfinity.org/t/internet-identity-deprecation-of-account-creation-on-all-origins-other-than-https-identity-ic0-app/9694
+ **/
+function isRegistrationAllowed() {
+  return !/(^https:\/\/rdmx6-jaaaa-aaaaa-aaadq-cai\.ic0\.app$)|(.+\.raw\..+)/.test(
+    window.origin
+  );
+}
+
+export const initRegistration = async (): Promise<LoginFlowResult | null> => {
+  const registerButton = document.getElementById(
+    "registerButton"
```

Comment:
this is quite dangerous, the function is expecting this element to be on the page but does not advertise this anywhere. how about only exporting the following, and let the caller bind it to any element/action they like?

```
isRegistrationAllowed() ? register() : registerDisabled()
```